connectors/sources/sharepoint_online.py

```diff
@@ -663,118 +719,259 @@ async def validate_config(self):
+                        site_list_name=site_list["name"],
                     ):
-                        list_item["_id"] = list_item["id"]
-                        list_item["object_type"] = "list_item"
-                        list_item["_tempfile_suffix"] = os.path.splitext(
-                            list_item.get("FileName", "")
-                        )[-1]
-
-                        content_type = list_item["contentType"]["name"]
-
-                        if content_type in [
-                            "Web Template Extensions",
-                            "Client Side Component Manifests",
-                        ]:  # TODO: make it more flexible. For now I ignore them cause they 404 all the time
-                            continue
-
-                        if "Attachments" in list_item["fields"]:
-                            async for list_item_attachment in self.client.site_list_item_attachments(
-                                site["webUrl"], site_list["name"], list_item["id"]
-                            ):
-                                list_item_attachment["_id"] = list_item_attachment[
-                                    "odata.id"
-                                ]
-                                list_item_attachment[
-                                    "object_type"
-                                ] = "list_item_attachment"
-                                list_item_attachment["_timestamp"] = list_item[
-                                    "lastModifiedDateTime"
-                                ]
-                                attachment_download_func = partial(
-                                    self.get_attachment_content,
-                                    list_item_attachment,
-                                )
-                                yield list_item_attachment, attachment_download_func
-
-                        download_func = None
-
                         yield list_item, download_func
 
-                async for site_page in self.client.site_pages(site["webUrl"]):
-                    site_page["_id"] = site_page["GUID"]
-                    site_page["object_type"] = "site_page"
+                # Sync site pages
+                async for site_page in self.site_pages(site["webUrl"]):
+                    yield site_page, None
 
-                    for html_field in ["LayoutWebpartsContent", "CanvasContent1"]:
-                        if html_field in site_page:
-                            site_page[html_field] = html_to_text(site_page[html_field])
+    async def get_docs_incrementally(self, sync_cursor, filtering=None):
+        self._sync_cursor = sync_cursor
 
-                    yield site_page, None
+        if self._sync_cursor is None or self._sync_cursor == {}:
+            raise SyncCursorEmpty(
+                "Incremental sync can't start because it doen't have information about the last full sync job. Please run a full sync to solve this."
```

Comment:
Idk - what if we make this message less specific just saying that "There's no cursor present"? E.g. not mentioning "full sync" as a reason for this, while still advising to run a full sync.

Reply:
I would like to avoid exposing such detailed implementation logic and expand the concepts a user needs to be aware of. _What is a cursor? Why would I, as a customer, care about cursors?_

I think a message like

> Incremental sync can't start because required information is missing. Please run a full sync to allow incremental syncs to run again.

would be useful.
cc: @leemthompo

Reply:
I agree with @danajuratoni.

Tiny copy tweak suggestion:

`Unable to start incremental sync. Please perform a full sync to re-enable incremental syncs.`